File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/fbo/attachment/Attachment.java
package org.saar.lwjgl.opengl.fbo.attachment;

import org.saar.lwjgl.opengl.fbo.attachment.allocation.AllocationStrategy;
import org.saar.lwjgl.opengl.fbo.attachment.buffer.AttachmentBuffer;
import org.saar.lwjgl.opengl.fbo.attachment.index.AttachmentIndex;

public class Attachment implements IAttachment {

    private final AttachmentBuffer buffer;
    private final AllocationStrategy allocation;

    public Attachment(AttachmentBuffer buffer, AllocationStrategy allocation) {
        this.buffer = buffer;
        this.allocation = allocation;
    }

    @Override
    public void init(AttachmentIndex index) {
        this.buffer.attachToFbo(index);
    }

    @Override
    public void allocate(int width, int height) {
        this.allocation.allocate(this.buffer, width, height);
    }

    @Override
    public void delete() {
        this.buffer.delete();
    }
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/fbo/attachment/Attachment.kt
package org.saar.lwjgl.opengl.fbo.attachment

import org.saar.lwjgl.opengl.fbo.attachment.allocation.AllocationStrategy
import org.saar.lwjgl.opengl.fbo.attachment.buffer.AttachmentBuffer
import org.saar.lwjgl.opengl.fbo.attachment.index.AttachmentIndex

class Attachment(
    private val buffer: AttachmentBuffer,
    private val allocation: AllocationStrategy
) : IAttachment {

    override fun attach(fbo: Int, index: AttachmentIndex) = this.buffer.attachToFbo(fbo, index)

    override fun allocate(width: Int, height: Int) = this.allocation.allocate(this.buffer, width, height)

    override fun delete() = this.buffer.delete()
}
